stock_prices/stock_prices.py

```diff
@@ -1,15 +1,54 @@
 #!/usr/bin/python
 
 import argparse
+import math
+import copy
+infinity = math.inf
+
+
+class Ledger:
+    __init__(self, buy=infinity, sell=-infinity, best=None):
+        self.buy = buy,
+        self.sell = sell,
+        self.money = sell_at(sell),
+        self.best = None
+
+    def sell_at(value):
+        self.sell = value
+        self.money = sell-buy
+
 
 def find_max_profit(prices):
-  pass
+
+    def recursively_profit(prices, money=Ledger()):
+        mo = money
+
+        for i, v in enumerate(prices):
+            if mo['money'] == -infinity:
+                if v < mo['buy']:
+                    mo['buy'] = v
+                elif v > mo['sell']:
+                    mo.sell_at(v)
+            elif mo['money'] != -infinity:
+                if v > mo['sell']:
+                    mo.sell_at(v)
+                elif v < mo['buy']:
+                    recursively_profit(prices[k:])
+
+    recursively_profit(prices)
```

Comment:
Im not quite sure what you are trying to do here, but we should not have a need for the Ledger as an extra input parameter to the function.  Nor does the recursive function return anything.  Consider using more meaningful names than "mo" and "v".

I would recommend coming back to these problems and the other problems in the repo as you get extra time to do so.